app/services/katello/pulp/smart_proxy_repository.rb

```diff
@@ -0,0 +1,179 @@
+module Katello
+  module Pulp
+    class SmartProxyRepository
+      attr_accessor :smart_proxy
+
+      def initialize(smart_proxy)
+        @smart_proxy = smart_proxy
+      end
+
+      def lifecycle_environments(organization_id = nil)
+        scope = @smart_proxy.lifecycle_environments
+        scope = scope.where(organization_id: organization_id) if organization_id
+        scope
+      end
```

Comment:
I think this method (or something similar) could be moved to the smart_proxy itself